### PR TITLE
Feature/turn time

### LINKED ISSUE #169: As a player, I want to see any timeouts that are running against me
# Task

When a timeout is running, a countdown should be displayed in the frontend, so I know that I have to react.
## Associated story

Story: #93

### LINKED ISSUE #180: Turntime can be set in the Blockchain
# Task

It is possible to set the timeout time in the Contract.
## Detailed Description

- init game accepts a time parameter
- getListOfGames return games with the time 
- move() resets the timeout
## Associated story

Story: #68

### PR DESCRIPTION
**Pull request by kyroy**
## Description

Add turn time to Frontend and Blockchain, without a clock for the user.

Pair programmed with @HeavyPanda 
## Related Issue

Working on #177
Needed for #178
Closes #180
Closes #169
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist contributor:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
## Checklist reviewer:
- [ ] The code has been peer-reviewed by @dennis-westphal

### CODE REVIEW

**Comment by graup** [comment]
This branch now also includes the countdown clock (next to the status). #169

Someone please review my last changes and merge :)

File: app/javascript/gamesFactory.js [hunk 1]
```diff
@@ -77,7 +77,7 @@ angular.module('dappChess').factory('games', function (crypto, navigation,
    *  pot: <number>,
    *  turnTime: <number>,
    *  timeoutStarted: <date>,
-   *  timeoutState: <{-2,-1,0,1,2}>
+   *  timeoutState: <{-1,0,1}>
```
Comment (dennis-westphal) [inline app/javascript/gamesFactory.js]: @graup: From what I can see in the contract, this is wrong. Any particular reason you changed it back from the (correct) state in develop?

File: app/javascript/gamesFactory.js [hunk 2]
```diff
@@ -74,15 +75,19 @@ angular.module('dappChess').factory('games', function (crypto, navigation,
    *  },
    *  ended: <boolean>,
    *  pot: <number>,
+   *  turnTime: <number>,
    *  timeoutStarted: <date>,
-   *  timeoutState: <{-2,-1,0,1,2}>
+   *  timeoutState: <{-1,0,1}>
```
Comment (dennis-westphal) [inline app/javascript/gamesFactory.js]: @graup: From what I can see in the contract, this is wrong. Any particular reason you changed it back from the (correct) state in develop?

Comment (graup) [inline app/javascript/gamesFactory.js]: No, must be a merge mistake. I took that line from @Kyroy @HeavyPanda 's branch, so I thought it should be right

**Comment by dennis-westphal** [comment]
There is something wrong about this. The turn time is display as timeout once a player used the "claim timeout" button, though in the description it says it should not be displayed at all. Also, the claimed timeout state does not seem to disappear, since the clock stays visible, yet the claim timeout ended button does not work. Also, the display of the timeout after it has run out is wrong.
screen shot 2016-07-11 at 10 57 03
screen shot 2016-07-11 at 10 57 17

**Comment by graup** [comment]
> [...] once a player used the "claim timeout" button [...]
> Also, the display of the timeout after it has run out is wrong.

It's intentional. Later the clients should send all claims (claimTimeout, claimTimeoutEnded) automatically. Implementing what happens in the client when the timer runs out was not part of this task. So instead of deliberately trying to clean up the output, I just left it as it is.

I would propose fixing these things after this PR is merged. This was just about adding the setting and displaying the time, not making the rest of the protocol work.

**Comment by dennis-westphal** [comment]
Ok. The code for the turn time looks good to me. The other bugs will be addressed in separate pull requests / commits.

**Pull request merged**